### ISSUE TITLE
Issues / Changelog script will need update 

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#10213]
cc @lexicalunit 

In light of the process changes that will come out of #10177, the `scripts/issues.py` tool will need updates. 

also ref: https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management

There are three modes that this script will be used:

#### Release Validation Mode

Would like to be able to run the script in automation to ensure that all the relevant issues are labeled appropriately, etc. Some criteria:

* Relevant issues/PRs are those for the current specified milestone
* All issues in the milestone should be closed, with a `reso:completed`
* All PRs in the milestone should be closed/merged with `status:accepted`
* Any PR with a `type` label (an "issue PR") also needs `reso:completed`
* There should only ever be one `reso` or `status` label applied.
* All issues sholud have a `type`
* Allowed types: bug, feature, task
* Nothing should have TRIAGE label
* All issues/PRs should be closed 

nice to have: verify that any plain PRs in the milestone link to an issue that is in the milestone, verify that for issues "closed by" a PR that that PR is in the milestone

```shell
python milestone.py 2.1.1  --check-only
```

#### Release Update Mode

Would like to be able to run the script in automation to update the CHANGELOG with lists of issues (or "issue PRs") from the current milestone. 

* Group by bugfix, feature, task
* Include component label, if present

```shell
python milestone.py 2.1.1  # Implies a check
```

**Comment by lexicalunit** [comment]
## Check Only Mode
- [ ] Relevant issues/PRs are those for the current specified milestone
- [ ] All issues in the milestone should be closed, with a reso:completed
- [ ] All PRs in the milestone should be closed/merged with status:accepted
- [ ] Any PR with a type label (an "issue PR") also needs reso:completed
- [ ] There should only ever be one reso or status label applied.
- [ ] All issues sholud have a type
- [ ] Allowed types: bug, feature, task
- [ ] Nothing should have TRIAGE label
- [ ] All issues/PRs should be closed

## Update Mode
- [ ] Should do all the checks from check only mode first
- [ ] Group by bugfix, feature, task
- [ ] Include component label, if present

**Comment by bryevdv** [comment]
closed by #10245
